src/main/java/com.gigssandbox/services/UserService.java

```diff
@@ -0,0 +1,55 @@
+package com.gigssandbox.services;
+
+import com.gigssandbox.entities.User;
+import com.gigssandbox.exceptions.AlreadyLoggedInException;
+import com.gigssandbox.exceptions.AlreadyRegisteredException;
+import com.gigssandbox.exceptions.IncorrectPasswordException;
+import com.gigssandbox.exceptions.NotRegisteredException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserService {
+    private final Map<String, User> users;
+
+    public UserService() {
+        this.users = new HashMap<>();
+    }
+
+    public void registerUser(String username, char[] password) throws AlreadyRegisteredException {
+        if (users.containsKey(username)) {
+            throw new AlreadyRegisteredException();
+        }
+        users.put(username, User.builder().username(username).passwordHash(Arrays.hashCode(password)).loggedIn(true).build());
```

Comment:
extract builder, this line is unreadable:

```
User user = User.builder()
                   .username(username)
                   .passwordHash(Arrays.hashCode(password))
                   .loggedIn(true)
                   .build();
users.put(username, user);
```

Reply:
and use CTRL+ALT+SHIFT+L to format it properly. For some reason github does not indent properly